assets/scripts/streets/SearchAddress.jsx

```diff
@@ -9,85 +11,125 @@ class SearchAddress extends Component {
   constructor (props) {
     super(props)
     this.state = {
-      value: ''
+      value: '',
+      placeholder: 'Enter Address',
+      suggestions: []
     }
 
-    this.onKeyUp = this.onKeyUp.bind(this)
+    this.throttleMakeRequest = throttle(this.makeRequest, 250)
+    this.handleSubmit = this.handleSubmit.bind(this)
+    this.getSuggestionValue = this.getSuggestionValue.bind(this)
+    this.onChange = this.onChange.bind(this)
+    this.renderSuggestion = this.renderSuggestion.bind(this)
+    this.onSuggestionsFetchRequested = this.onSuggestionsFetchRequested.bind(this)
     this.handleJSON = this.handleJSON.bind(this)
     this.searchAddress = this.searchAddress.bind(this)
   }
 
-  onKeyUp (e) {
+  onSuggestionsFetchRequested (x) {
+    if (x.value.length >= 2) {
+      const searchUrl = `${apiurl}?api_key=${apikey}&text=${x.value}`
+      window.fetch(searchUrl).then(this.searchAddress)
+    }
+  }
+
+  onChange (e, x) {
     this.setState({
-      value: e.target.value
+      value: x.newValue
     })
-
-    this.props.setMapState({ rawInputString: this.state.value })
   }
 
   handleJSON (res, err) {
     this.setState({
-      loading: false
+      suggestions: res.features
     })
-
     if (res.features === undefined || res.features.length === 0) {
       return
     }
 
-    this.props.searchResults(res.features[0].geometry.coordinates.reverse(), this.props.addressInformationLabel)
+    this.setState({
+      coordReverse: res.features[0].geometry.coordinates.reverse(),
```

Comment:
Regarding the first comment, yes. It seems this causes a selected item to always use the first result even though someone selected a different one.

We definitely should make use of `onSuggestionSelected` to handle the selected menu item.